app/src/main/java/com/example/fishjamandroidexample/CameraService.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.fishjamandroidexample
+
+import android.app.ForegroundServiceStartNotAllowedException
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.content.pm.ServiceInfo
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import androidx.core.app.ServiceCompat
+import androidx.core.content.ContextCompat
+
+class CameraService : Service() {
+    private fun startForeground() {
+        // Before starting the service as foreground check that the app has the
+        // appropriate runtime permissions. In this case, verify that the user has
+        // granted the CAMERA permission.
+        val cameraPermission =
+            ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA)
+        if (cameraPermission == PackageManager.PERMISSION_DENIED) {
+            // Without camera permissions the service cannot run in the foreground
+            // Consider informing user or updating your app UI if visible.
+            stopSelf()
+            return
+        }
+
+        try {
+            val channelID = "FISHJAM_NOTIFICATION_CHANNEL"
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
I spent too much time figuring out what is `VERSION_CODES.(zero)` 🙈 
_Before I realised it is letter `o`_